src/cls-ah.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const asyncHook = require('async_hooks');
+
+const wrappedSymbol = Symbol('context_wrapped');
+let contexts = {};
+let current = {};
+
+const hook = asyncHook.createHook({init, before, destroy});
+hook.enable();
+
+function Namespace() {}
+
+Namespace.prototype.get = function(k) {
+  return current ? current[k] : undefined;
```

Comment:
When would current be `null`?

Reply:
This can no longer be the case. Removed the checks.